x/storage/keeper/abci.go

```diff
@@ -38,11 +39,22 @@ func EndBlocker(ctx sdk.Context, keeper Keeper) {
 	}
 
 	// delete buckets
-	_, err = keeper.DeleteDiscontinueBucketsUntil(ctx, blockTime, deletionMax-deleted)
-	if err != nil {
-		ctx.Logger().Error("should not happen, fail to delete buckets, err " + err.Error())
-		panic("should not happen")
+	doDeleteBucket := true
+	if ctx.BlockHeight() > 5946511 && ctx.ChainID() == "greenfield_5600-1" {
```

Comment:
Shall we add some comments here on why we need this?

Reply:
added